### PR TITLE
pixijs v4.3 and pixi-spine updates

### PR DESCRIPTION
**Pull request by ivanpopelyshev**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `npm run new-package package-name`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by clark-stevenson** [comment]
👍 Awesome Ivan

File: pixi.js/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
-// Type definitions for Pixi.js v4.3.0
-// Project: https://github.com/pixijs/pixi.js/
-// Definitions by: Clark Stevenson <https://github.com/clark-stevenson>
+// Type definitions for Pixi.js 4.3.4
```
Comment (DanielRosenwasser) [inline pixi.js/index.d.ts]: Can you keep this at 4.3? We don't keep anything other than major & minor version numbers.

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]
Can you also add a `tslint.json`?

File: pixi.js/index.d.ts (hunk 1 above)
Comment (ivanpopelyshev) [inline pixi.js/index.d.ts]: Yeah, no problem. People can use special compatibility plugin that adds new API to older versions of lib, so this definition will work even for any 4.x

**Comment by ivanpopelyshev** [comment]
@DanielRosenwasser I didn't use tslint yet, and we don't have code conventions for pixi-typescript :(

**Comment by ivanpopelyshev** [comment]
Thanks to that tslint it'll be merged later.

**Comment by clark-stevenson** [comment]
@DanielRosenwasser

I can't give you a tslint right now.  There are problems, this is one of them.... but not really because the real problem is.

1. How do I generate d.ts from a JS codebase? I don't mean that in a silly way.  Is there any documentation systems which allow me to describe my intent?
2. I re-write the project in TypeScript and maintain that. 

Until one of those are done, this is, and always will be a hackjob, but it works, and it is either that or `any`.

**Comment by aozgaa** [comment]
regarding tslint, see this for advice: https://github.com/DefinitelyTyped/DefinitelyTyped#lint

The `../tslint.json` is the tslint file in the root directory of the repo. It contains our default rules.

**Comment by clark-stevenson** [comment]
I still can't see that `../tslint.json` and if `pixi-typescript` is going to pass linting, then I need the same rules as DT.

**Comment by aozgaa** [comment]
If you have the repo cloned, and then running
```
npm install tslint typescript -g
tslint -c /path/to/tslint.json /path/to/DT/pixi.js/*.ts
```

should work. Are you running into issues there?

**Comment by ivanpopelyshev** [comment]
@aozgaa I dont understand what are you doing :)

**Comment by GeorgeA93** [comment]
Any chance of an update on the status of this PR?

**Comment by ivanpopelyshev** [comment]
1. we have to exclude "v3" folder from lint.
2. in index ```2867:26   variable-name   variable name clashes with keyword/type```

**Comment by clark-stevenson** [comment]
That second one needs `"variable-name": false` as `isMobile.any` is a valid property of the library.

**Comment by ivanpopelyshev** [comment]
All failed and no log. What's happening ?

**Comment by ivanpopelyshev** [comment]
@DanielRosenwasser Quick! Merge it while its green!

**Comment by samalexander** [comment]
@DanielRosenwasser  Please can we get this merged?

**Comment by aozgaa** [comment]
Please follow the schema for `tslint.json` files. You don't need any `jsRules` since there are not `*.js` files in the directory. I cloned your repo and this should work:
```
{
    "extends": [
        "../tslint.json"
    ],
    "rules": {
        "forbidden-types": false,
        "no-single-declare-module": false,
        "unified-signatures": false,
        "interface-name": [false],
        "max-line-length": [false]
    }
}
```

**Review by aozgaa — CHANGES_REQUESTED** [review]
Please follow the schema for `tslint.json` files. You don't need any `jsRules` since there are not `*.js` files in the directory. I cloned your repo and this should work:
```
{
    "extends": [
        "../tslint.json"
    ],
    "rules": {
        "forbidden-types": false,
        "no-single-declare-module": false,
        "unified-signatures": false,
        "interface-name": [false],
        "max-line-length": [false]
    }
}
```

**Comment by ivanpopelyshev** [comment]
It doesn't work. I see a lot of errors from index and tests.

```
d:\pixi-games\DefinitelyTyped>npm run lint pixi.js

> definitely-typed@0.0.1 lint d:\pixi-games\DefinitelyTyped
> node scripts/lint.js "pixi.js"

node ../node_modules/tslint/lib/tslint-cli --format stylish "**/*.ts"
index.d.ts
1597:9    adjacent-overload-signatures  All 'from' signatures should be adjacent
1083:20   array-type                    Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead.
...
```

**Comment by clark-stevenson** [comment]
Can anyone help get this merged?

I just want to underline the situation in case it got lost.  We can mess around with this PR for another 5-6 weeks, but it don't resolve the underlying issue that this PR is coming from `pixi-typescript` and that no one can use the project because it is has been stuck here for weeks.

As soon as this PR is accepted, I am going to be taking the last few weeks of `pixi-typescript` updates and making a PR with pretty much all of the same problems all over again.  There is zero parity between this PR and that project.

If we can't merge this asap, then DT is not the answer, and I desperately need to know how to give the people who keep asking me a way to acquire the project.  I am quickly losing all hope and interest of maintaining pixi declarations which is tedious by itself, and this 5-6 week PR system blocking anyone from using it is destroying the purpose of even trying. 

Far too much Red Tape.

**Comment by mhegazy** [comment]
sorry for the delay.

**Comment by clark-stevenson** [comment]
@mhegazy not at all! Thank you very much for your time!

**Pull request merged**